sycl/plugins/opencl/pi_opencl.hpp

```diff
@@ -52,12 +51,25 @@ class OpenCLVersion {
      * 'OpenCL<space><ocl_major_version.ocl_minor_version><space><vendor-specific
      * information>' for devices.
      */
-    std::regex rx("OpenCL ([0-9]+)\\.([0-9]+)");
-    std::smatch match;
+    std::string_view prefix = "OpenCL ";
+    size_t versionBegin = version.find_first_of(" ");
+    size_t versionEnd = version.find_first_of(" ", versionBegin + 1);
+    size_t versionSeparator = version.find_first_of(".", versionBegin + 1);
 
-    if (std::regex_search(version, match, rx) && (match.size() == 3)) {
-      ocl_major = strtoul(match[1].str().c_str(), nullptr, 10);
-      ocl_minor = strtoul(match[2].str().c_str(), nullptr, 10);
+    bool haveOCLPrefix =
+        std::equal(prefix.begin(), prefix.end(), version.begin());
+
+    if (haveOCLPrefix && versionBegin != std::string::npos &&
```

Comment:
I wonder if there is any situation where this is expected to fail at runtime? If not, we could maybe turn this if condition into an assertion

Reply:
From what I found of how this function was used. Is that if an invalid string version was passed, it will keep the `ocl_major` - `ocl_minor` with a 0 value, which `isValid` function will return false in that case. and how I saw this propagating up. Is that it will return a relevant cl error code associated with the called function like `CL_INVALID_PLATFORM` in `getPlatformVersion`. I think the current way will give more information of where the faulty version string was passed from, but asserting will bail out the program without giving this information. however, I think this is more still of an assertion situation, but not sure if the assertion way could make some resources not being deleted appropriately or this cl error code are more of expected thing in the source code in another code place that depend on it.

Reply:
Sorry, I forgot to answer this. What you say makes sense. I think it's fine to keep it as is :+1: